homeassistant/components/incomfort/__init__.py

```diff
@@ -0,0 +1,57 @@
+"""Support for an Intergas boiler via an InComfort/Intouch Lan2RF gateway."""
+import logging
+
+import voluptuous as vol
+from incomfortclient import Gateway as InComfortGateway
+
+from homeassistant.const import (
+    CONF_HOST, CONF_PASSWORD, CONF_USERNAME)
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.discovery import async_load_platform
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'incomfort'
+
+_V1_SCHEMA = vol.Schema({
+    vol.Required(CONF_HOST): cv.string,
+})
+_V2_SCHEMA = vol.Schema({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Required(CONF_USERNAME): cv.string,
```

Comment:
Looks like we can simplify and only keep one schema and use vol.Inclusive on the optional credentials.
```suggestion
    vol.Inclusive(CONF_USERNAME, 'creds'): cv.string,
```

Reply:
Sorted: 
```python
CONFIG_SCHEMA = vol.Schema({
    DOMAIN: vol.Schema({
        vol.Required(CONF_HOST): cv.string,
        vol.Inclusive(CONF_USERNAME, 'credentials'): cv.string,
        vol.Inclusive(CONF_PASSWORD, 'credentials'): cv.string,
    })
}, extra=vol.ALLOW_EXTRA)
```